cmd/osv-scanner/main.go

```diff
@@ -113,15 +117,15 @@ func run(args []string, stdout, stderr io.Writer) int {
 				format = "json"
 			}
 
-			switch format {
-			case "json":
-				r = reporter.NewJSONReporter(stdout, stderr)
-			case "table":
-				r = reporter.NewTableReporter(stdout, stderr, false)
-			case "markdown":
-				r = reporter.NewTableReporter(stdout, stderr, true)
-			default:
-				return fmt.Errorf("%v is not a valid format", format)
+			outputPath := context.String("output")
```

Comment:
Any reason to create an in-memory buffer instead of directly writing to the file ?

Reply:
Because this way it only writes it to file when osv-scanner succeeds, if it's writing directly to file it'll always create a file, but upon an error (e.g. the file it's pointed to scan doesn't exit, the flags passed in are wrong...etc) it'll exit early and nothing will be written to it. 

Though now thinking about it, it might be surprising behavior, maybe it should just create an empty file in those cases.

Reply:
Yeah, creating an empty file seems like reasonable behaviour. Would we still need this buffer in that case? 

We could also alternatively just clear the file on an error to avoid this indirection with the buffer in that case?